ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/typemodels/BaseTypeGenerator.java

```diff
@@ -0,0 +1,273 @@
+        if (isEmpty(annotation.getDs3AnnotationElements())
+                || !annotation.getName().endsWith("CustomMarshaledName")) {
+            return "";
+        }
+        for (final Ds3AnnotationElement annotationElement : annotation.getDs3AnnotationElements()) {
+            if (annotationElement.getName().equals("Value")) {
+                return annotationElement.getValue();
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Converts a list of Ds3EnumConstants into a list of EnumConstant models
+     * @param ds3EnumConstants A list of Ds3EnumConstants
+     * @return A list of EnumConstant models
+     */
+    protected ImmutableList<EnumConstant> toEnumConstantList(
+            final ImmutableList<Ds3EnumConstant> ds3EnumConstants) {
+        if (isEmpty(ds3EnumConstants)) {
+            return ImmutableList.of();
+        }
+        final ImmutableList.Builder<EnumConstant> builder = ImmutableList.builder();
+        for (final Ds3EnumConstant ds3EnumConstant : ds3EnumConstants) {
+            builder.add(toEnumConstant(ds3EnumConstant));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Converts a Ds3EnumConstant into an EnumConstant model
+     * @param ds3EnumConstant A Ds3EnumConstant
+     * @return An EnumConstant model
+     */
+    protected static EnumConstant toEnumConstant(final Ds3EnumConstant ds3EnumConstant) {
+        return new EnumConstant(ds3EnumConstant.getName());
+    }
+
+    /**
+     * Gets all the required imports that the Model will need in order to properly
+     * generate the java model code
+     */
+    protected ImmutableList<String> getAllImports(final Ds3Type ds3Type) {
+        //If this is an enum, then there are no imports
+        if (hasContent(ds3Type.getEnumConstants())) {
+            return ImmutableList.of();
+        }
+        return getImportsFromDs3Elements(ds3Type.getElements());
+    }
+
+    /**
+     * Gets all the required imports that the elements will need in order to properly
+     * generate the java model
+     */
+    protected static ImmutableList<String> getImportsFromDs3Elements(final ImmutableList<Ds3Element> elements) {
+        if (isEmpty(elements)) {
+            return ImmutableList.of();
+        }
+        final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        for (final Ds3Element element : elements) {
+            if (element.getType().contains(".")) {
+                builder.add(ConvertType.toModelName(element.getType()));
+            }
+            if (hasContent(element.getComponentType())
+                    && element.getComponentType().contains(".")) {
+                builder.add(ConvertType.toModelName(element.getComponentType()));
+                builder.add("java.util.List");
+                builder.add("com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper");
+            }
+        }
+        return builder.build().asList();
```

Comment:
This function is creating a list of models to import. Since its possible that multiple Ds3Elements use the same type, I only want each import of said type once. I'm using the set to remove duplicates.